enterprise/backend/src/metabase_enterprise/serialization/api.clj

```diff
@@ -0,0 +1,237 @@
+      (doseq [^TarArchiveEntry e (entries tar)]
+        (when-not @dir-name
+          (reset! dir-name (.getName e)))
+        (let [f (io/file dst (.getName e))]
+          (if (.isFile e)
+            (io/copy tar f)
+            (.mkdirs f)))))
+    @dir-name))
+
+;;; Logging
+
+(defn- find-logger-layout
+  "Find any logger with a specified layout"
+  [^LoggerConfig logger]
+  (when logger
+    (or (first (keep #(.getLayout ^AbstractAppender (val %)) (.getAppenders logger)))
+        (recur (.getParent logger)))))
+
+(defn- make-logger ^AutoCloseable [ns ^File f]
+  (let [config        (logger/configuration)
+        parent-logger (logger/effective-ns-logger ns)
+        appender      (.build
+                       (doto (FileAppender/newBuilder)
+                         (.withName (.getPath f))
+                         (.withFileName (.getPath f))
+                         (.withLayout (find-logger-layout parent-logger))))
+        logger        (LoggerConfig. (logger/logger-name ns) Level/INFO true)]
+    (.start appender)
+    (.addAppender config appender)
+    (.addAppender logger appender (.getLevel logger) nil)
+    (.addLogger config (.getName logger) logger)
+    (.updateLoggers (logger/context))
+
+    (reify AutoCloseable
+      (close [_]
+        (let [^AbstractConfiguration config (logger/configuration)]
+          (.removeLogger config (.getName logger))
+          (.stop appender)
+          ;; this method is only present in AbstractConfiguration
+          (.removeAppender config (.getName appender))
+          (.updateLoggers (logger/context)))))))
+
+;;; Logic
+
+(defn- serialize&pack ^File [opts]
+  ;; TODO: how to get time in local timezone?
+  (let [id       (format "%s-%s"
+                         (u/slugify (public-settings/site-name))
+                         (u.date/format "YYYY-MM-dd_HH-mm" (t/zoned-date-time)))
+        path     (io/file parent-dir id)
+        dst      (io/file (str (.getPath path) ".tar.gz"))
+        log-file (io/file path "export.log")]
+    (with-open [_logger (make-logger 'metabase-enterprise.serialization log-file)]
+      (try                              ; try/catch inside logging to log errors
+        (-> (extract/extract opts)
+            (storage/store! path))
+        ;; not removing storage immediately to save some time before response
+        (compress-tgz path dst)
+        (catch Exception e
+          (log/error e "Error during serialization"))))
+    {:archive  (when (.exists dst)
+                 dst)
+     :log-file (when (.exists log-file)
+                 log-file)
+     :callback (fn []
+                 (when (.exists path)
+                   (run! io/delete-file (reverse (file-seq path))))
+                 (when (.exists dst)
+                   (io/delete-file dst)))}))
+
+(defn- unpack&import [^File file & [size]]
+  (let [dst      (io/file parent-dir (apply str (repeatedly 20 #(char (+ 65 (rand-int 26))))))
```

Comment:
> (apply str (repeatedly 20 #(char (+ 65 (rand-int 26))

we have `(mt/random-name)` that does exactly this, maybe we should extract it to `metabase.util` so we can reuse it?

Reply:
I've been thinking about that back and forth... but yeah, why not 😁